lib/core/realtime/notifier.js

```diff
@@ -99,276 +173,206 @@ class NotifierController {
       scope
     });
 
-    return this._notifyUser(
-      room,
-      request,
-      scope,
-      content,
-      { fromCluster: false });
+    return this._notifyUser(room, request, scope, content, {
+      fromCluster: false
+    });
   }
 
   /**
-   * Send a server notification to a provided connection identifier
+   * Send a "token expired" notification to the target user
    *
-   * @param {Array} rooms - User's rooms to notify
    * @param {string} connectionId - User's connection identifier
-   * @param {string} type - Server notification type
-   * @param {string} message - Additional information
-   *
    * @returns {Promise}
    */
-  async notifyServer (rooms, connectionId, type, message) {
+  async notifyTokenExpired (connectionId) {
+    const rooms = this.module.hotelClerk.getUserRooms(connectionId);
+
+    if (rooms.length === 0) {
+      return;
+    }
+
     const channels = [];
 
-    if (rooms.length > 0) {
-      for (const room of rooms) {
-        const hotelClerkRoom = this.kuzzle.hotelClerk.rooms.get(room);
+    for (const room of rooms) {
+      const hotelClerkRoom = this.module.hotelClerk.rooms.get(room);
 
-        if (hotelClerkRoom !== undefined) {
-          channels.push(...Object.keys(hotelClerkRoom.channels));
-        }
+      if (hotelClerkRoom !== undefined) {
```

Comment:
Nitpicking
```suggestion
      if (hotelClerkRoom) {
```

Reply:
Since we get a value from a map, I think it's perfectly valid to check against `undefined` to test whether the value exists.
I don't think I'll apply this nitpicking, unless you've a strong argument in its favor.

Reply:
It's just because it's more concise and we are not suppose to store `false` or `null` values inside this map but it was really a small nitpicking and also I agree that this is the actual pattern when checking a value get from a map :+1: